### PR TITLE
Remove WeakReference tracking from PendingTrace

### PR DESCRIPTION
**Pull request by tylerbenson**
This means we rely on the delay from PendingTraceBuffer to report traces where the root span is finished but the pendingReference count is not 0. No more cleaning required.

### CODE REVIEW

File: dd-trace-core/src/main/java/datadog/trace/core/PendingTrace.java [hunk 1]
```diff
@@ -250,30 +201,16 @@ long oldestFinishedTime() {
...
-        continuation.cancel(weakContinuations);
-        expireReference(false);
-      } else {
-        log.debug("t_id={} -> not registered in trace: {}", traceId, continuation);
-      }
-    }
+    expireReference(false);
   }
 
   private void expireReference(boolean isRootSpan) {
```
Comment (dougqh) [inline dd-trace-core/src/main/java/datadog/trace/core/PendingTrace.java:216]: Now that there is no reference, should we change the name of `expiredReference`?

**Comment by dougqh** [comment]
Overall, I think this looks good.

My main question is how we're testing this.  I suppose the tests added in the recent PR are probably sufficient, but I think it would be good to clarify.

We should also check the impact especially on memory.  Since we've eliminated the reference objects & queues, I expect this to better.  However, I'd like us to be in the habit of double checking.

File: dd-trace-core/src/main/java/datadog/trace/core/PendingTrace.java (hunk 1 above)
Comment (tylerbenson) [reply dd-trace-core/src/main/java/datadog/trace/core/PendingTrace.java]: I renamed it, but not in love with the new name.

File: .palantir/revapi.yml [hunk 2]
```diff
@@ -759,6 +759,12 @@ acceptedBreaks:
       old: "method E java.util.concurrent.ConcurrentLinkedDeque<E>::removeLast() @\
         \ datadog.trace.core.PendingTrace"
       justification: "encapsulate PendingTrace better"
+    - code: "java.method.removed"
```
Comment (richardstartin) [inline .palantir/revapi.yml:762]: let's bin revapi already

**Review by richardstartin — APPROVED** [review]

**Review by bantonsson — APPROVED** [review]

**Pull request merged**